src/source_frames.rs

```diff
@@ -251,18 +266,47 @@ fn drop_frame_tc_adjustment(sections: TimecodeSections, rate: Framerate) -> Fram
     Ok(-adjustment)
 }
 
-fn parse_feet_and_frames_str(matched: regex::Captures) -> FramesSourceResult {
+fn parse_feet_and_frames_str(
+    matched: regex::Captures,
+    given_format: Option<FilmFormat>,
+) -> FramesSourceResult {
     // If we got a match, these groups had to be present, so we can unwrap them.
+
     let feet = timecode_parse::convert_tc_int(matched.name("feet").unwrap().as_str(), "feet")?;
-    let mut frames =
+    let frames =
         timecode_parse::convert_tc_int(matched.name("frames").unwrap().as_str(), "frames")?;
 
+    let perfs_str = matched.name("perf");
+
     // Get whether this value was a negative timecode value.
     let is_negative = matched.name("negative").is_some();
-
-    frames += feet * FRAMES_PER_FOOT;
-    if is_negative {
-        frames = -frames;
+    let perfs_n = perfs_str.and_then(|x| x.as_str().parse::<i64>().ok());
+
+    let final_format : Result<FilmFormat, TimecodeParseError> = match (given_format, perfs_n) {
+        (Some(x) , Some(_)) if !x.allows_perf_field()  => Err(TimecodeParseError::UnknownStrFormat(
+             format!("Perf field was present in string \"{}\", which is not allowed for given film format {:?}.", 
+                 matched.get(0).unwrap().as_str(), x)
+             )
+            ),
+        (Some(f), _) => Ok(f),
+        (None, Some(_)) => Ok(FilmFormat::FF35mm3perf),
+        (_, _) => Ok(FilmFormat::FF35mm4perf),
     };
-    Ok(frames)
+
+    if let Ok(ff) = final_format {
+        let int_feet = (feet / ff.footage_modulus()) * ff.footage_modulus();
```

Comment:
I'm a little confused here, if you're dividing feet by a value, then immediately multiplying by the same, shouldn't you just end up where you started? What is the purpose of this step?

Reply:
Also I think you are using `int` here to mean `integral`, rather than `integer`, yeah? I think that should probably be fully written out to disambiguate it.

Reply:
I think I just need to comment this section heavily, basically I'm just getting the floor footage for this module... I'll type out the notes.